### PR TITLE
[BugFix] Be may crash after upgrading from 2.4.0-rc version to 2.4.0

### LINKED ISSUE #13126: [Bug] version-2.4.0 and rc versions have compatibility issues
### Steps to reproduce the behavior (Required)
1. CREATE TABLE using persistent index in rc version 
2. upgrade to 2.4.0
3. be may be crash

### Expected behavior (Required)
be run success

### Real behavior (Required)
Be may crash in 2.4.0
image
Be crash because of  divide by 0 error, but this should not happened. 
The real reason is the compatibility
image
The above figure shows the ShardInfo of L1,  the scenario of data_size being 0 but size is not 0 should not happen and it will cause the crash.

The reason is we add `data_size` into ShardInfo in version 2.4.0, so when we upgade from rc version, the `data_size` will be treated as default vaule which is 0.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/13126

## Problem Summary(Required) ：

In order to fix the bug(https://github.com/StarRocks/starrocks/issues/11868), we add `data_size` into `ShardInfo` proto, but this may cause some compatibility problem and Be maybe crash after upgrade to 2.4.0.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Review by chaoyli — APPROVED** [review]

File: be/src/storage/persistent_index.cpp [hunk 1]
```diff
@@ -2282,7 +2282,17 @@ StatusOr<std::unique_ptr<ImmutableIndex>> ImmutableIndex::load(std::unique_ptr<R
...
         dest.nbucket = src.nbucket();
-        dest.data_size = src.data_size();
+        // This is for compatibility, we don't add data_size in shard_info in the rc version
+        // And data_size is added to reslove some bug(https://github.com/StarRocks/starrocks/issues/11868)
+        // However, if we upgrade from rc version, the data_size will be used as default value(0) which will cause
+        // some error in the subsequent logic
+        // So we will use file size as data_size which will cause some of disk space to be wasted, but it is a acceptable
+        // problem. And the wasted disk space will be reclaimed in the subsequent compaction, so it is acceptable
+        if (src.size() != 0 && src.data_size() == 0) {
+            dest.data_size = src.data().size();
```
Comment (trueeyu) [inline be/src/storage/persistent_index.cpp:2292]: dest.data_size = 0;  ??

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
@mergify backport branch-2.4

**Review by trueeyu — APPROVED** [review]

**Comment by sevev** [comment]
@mergify backport branch-2.5

**Pull request merged**